lib/SIL/IR/SILWitnessTable.cpp

```diff
@@ -171,11 +171,19 @@ bool SILWitnessTable::conformanceIsSerialized(
   if (normalConformance && normalConformance->isResilient())
     return false;
 
-  if (conformance->getProtocol()->getEffectiveAccess() < AccessLevel::Public)
+  // Allow serializing conformance with package access level if
+  // package serialization is enabled.
+  auto optInPackage = conformance->getDeclContext()
+                          ->getASTContext()
+                          .SILOpts.EnableSerializePackage;
+  auto accessLevelToCheck =
+      optInPackage ? AccessLevel::Package : AccessLevel::Public;
+
+  if (conformance->getProtocol()->getEffectiveAccess() < accessLevelToCheck)
```

Comment:
Without this, witness thunks for conformance with package linkage don't get proper linkages; see SILGenConformance::addMethodImplementation(..)